File: GFG/Arrays/majority.py
def majority(arr):
    res=0
    count=1
    n=len(arr)
    for i in range(1,n):
        if arr[res]==arr[i]:
            count+=1
        else:
            count-=1
        if count==0:
            res=i
            count=1
    count=0
    for i in range(n):
        if arr[res]==arr[i]:
            count+=1
    if count<n//2:
        return -1
    return res

brr=[8,8,8,8,7,7,8,6,6,8,6]
print(majority(brr))

File: GFG/Arrays/majority.cpp
#include<iostream>
#include<vector>
using namespace std;

int majority(vector <int> &arr){
    int n=arr.size();
    int res=0;
    int count=1;
    for(int i=1;i<n;i++){
        if(arr[res]==arr[i]){
            count++;
        }
        else{count--;}
        if (count<=0){res=i;count=1;}
    }
    count=0;
    for(int i=0;i<n;i++){
        if(arr[res]==arr[i]){
            count++;
        }
    }
    if(count<n/2){
        return -1;
    }
    return res;
}

int main(){
    vector<int> arr={8,8,6,6,6,4,6};
    cout<<majority(arr);
    return 0;
}